sql/src/test/java/org/apache/druid/sql/calcite/CalciteJoinQueryTest.java

```diff
@@ -3826,13 +3826,16 @@ public void testJoinWithExplicitIsNotDistinctFromCondition(Map<String, Object> q
                 .context(queryContext)
                 .build()
         ),
-        ImmutableList.of(
-            new Object[]{"", ""},
-            new Object[]{"10.1", "10.1"},
-            new Object[]{"2", "2"},
-            new Object[]{"1", "1"},
-            new Object[]{"def", "def"},
-            new Object[]{"abc", "abc"}
+        sortIfSortBased(
+            ImmutableList.of(
+                new Object[]{"", ""},
+                new Object[]{"10.1", "10.1"},
+                new Object[]{"2", "2"},
+                new Object[]{"1", "1"},
+                new Object[]{"def", "def"},
+                new Object[]{"abc", "abc"}
+            ),
+            0
         )
     );
   }
```

Comment:
Good news, `testJoinWithExplicitIsNotDistinctFromCondition` is that test case 😄

This test was passing on `master` because the join was getting switched quietly to broadcast. Now it actually runs as sort-merge, which is why I had to add `sortIfSortBased` (the results are now in a different order).